lib/src/tag/_common.dart

```diff
@@ -16,6 +16,26 @@ typedef Explainer<N> = Explanation? Function(List<N> path);
 /// Explainer should give an explanation of the problem.
 typedef Predicate<T> = Explainer<T>? Function(T node);
 
+/// The accumulated values of a tagging step.
+class TaggingResult {
+  final Set<String> tags;
+  final List<Explanation> explanations;
+
+  TaggingResult(this.tags, this.explanations);
+
+  factory TaggingResult.merge(Iterable<TaggingResult> results) {
+    final tags = <String>{};
+    final explanations = <Explanation>[];
+    for (final result in results) {
+      tags.addAll(result.tags);
+      explanations.addAll(result.explanations);
```

Comment:
Won't this give conflicting tags and explanations?

Like an explanation saying "Doesn't support linux because", while it has the tag `platform:linux`

Reply:
The outer loop iterates over the platforms, and for each of them, we put the dart and flutter runtime tagging result into a list (if they are available, see tagger.dart#307). If any of them is a success, we award the tag, and don't expose any explanation (if the other runtime would fail). Otherwise we return the first result (which like has a failure explanation).

Reply:
Moved the inner method to class-level, it makes this logic more apparent, PTAL.